cime_config/cesm/machines/Makefile

```diff
@@ -291,14 +288,7 @@ endif
 #===============================================================================
 # Set include paths (needed after override for any model specific builds below)
 #===============================================================================
-INCLDIR += -I$(SHAREDPATH)/include -I$(SHAREDPATH)/$(COMP_INTERFACE)/$(ESMFDIR)/$(NINST_VALUE)/csm_share
-
-#
-#  Tell gmake to look in the csm_share directory for shr_*.mod files
-#
-
-vpath  shr%.mod $(SHAREDPATH)/$(COMP_INTERFACE)/$(ESMFDIR)/$(NINST_VALUE)/csm_share
-
+INCLDIR += -I$(INSTALL_SHAREDPATH)/include -I$(INSTALL_SHAREDPATH)/$(COMP_INTERFACE)/$(ESMFDIR)/$(NINST_VALUE)/include
```

Comment:
If I'm understanding this change (and other similar changes) correctly, it means that, even for the shared library builds, the include directory is pointing to the "install" location - i.e., some case-specific location. That implies that (e.g.) the sharedlib csm_share build is done against a case-specific installation of the mct build. I'm not sure if this will cause problems in practice, although it feels weird.

I had imagined that anything built in the sharedlib area would be built against module and .h files in the sharedlib area, in order to avoid having things from sharedlib point into specific cases. Then it would just be the things built in the case-specific location (i.e., non-shared-lib) that would point to this new INSTALL_SHAREDPATH.

I don't know how important this is, though, beyond a general sense of aesthetics. Is it important to avoid problems if the case-specific install directory is blown away???